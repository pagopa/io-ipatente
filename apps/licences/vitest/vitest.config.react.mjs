/// <reference types="vitest" />
/// <reference types="vite/client" />

import react from "@vitejs/plugin-react";
import { loadEnv } from "vite";
import tsconfigPaths from "vite-tsconfig-paths";
import { configDefaults, defineConfig } from "vitest/config";

export default defineConfig({
  plugins: [react(), tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...configDefaults.coverage.exclude,
        "**/__tests__",
        "src/generated/**",

        // BE
        "src/app/*",
        "src/config/*",
        "src/lib/bff/*",
      ],
      ignoreEmptyLines: true,
      include: ["src/**/*.{tsx,ts}"],
      reporter: ["text", "html"],
    },
    css: true,
    env: loadEnv("test", process.cwd(), ""),
    environment: "jsdom",
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    globals: true,
    include: ["./src/**/*.test.tsx"],
    setupFiles: "./vitest/vitest.setup.react.ts",
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
