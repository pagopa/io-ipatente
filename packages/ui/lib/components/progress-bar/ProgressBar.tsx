import { LinearProgress, Stack, Typography } from "@mui/material";

export interface ProgressBarProps {
  description?: string;
  title?: string;
  total: number;
  value: number;
}
export const ProgressBar = ({
  description,
  title,
  total,
  value,
}: ProgressBarProps) => (
  <Stack rowGap="8px" width={"100%"}>
    <Stack direction="row" justifyContent="space-between">
      {title && (
        <Typography
          data-testid="progress-bar-title"
          fontSize={20}
          fontWeight={600}
        >
          {title}
        </Typography>
      )}
      {description && (
        <Typography
          data-testid="progress-bar-description"
          fontSize={20}
          fontWeight={600}
        >
          {description}
        </Typography>
      )}
    </Stack>
    <LinearProgress
      sx={{
        borderRadius: "16px",
        height: "8px",
      }}
      value={(value / total) * 100}
      variant="determinate"
    />
  </Stack>
);
