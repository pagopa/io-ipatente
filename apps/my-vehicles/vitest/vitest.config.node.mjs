import { configDefaults, defineConfig } from "vitest/config";
import tsconfigPaths from "vite-tsconfig-paths";

export default defineConfig({
  plugins: [tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...(configDefaults.coverage.exclude
          ? configDefaults.coverage.exclude
          : []),
        ".next/*",
        "mocks/*",
        "public/*",
        "*.config.js",
        "src/generated/**",
        "src/config/*",
        "src/types/*",
        "src/main.ts",
        "src/config.ts",
        "src/instrumentation.ts",
        "**/__mocks__/**",
        // FE
        "src/hooks/*",
        "src/layouts/*",
      ],
      extension: configDefaults.coverage.extension
        ?.toString()
        .replace(",.tsx", "")
        .split(","),
      ignoreEmptyLines: true,
      reporter: ["text", "json-summary", "json"],
      reportsDirectory: `${configDefaults.coverage.reportsDirectory}/backend`,
    },
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    include: ["./src/**/__tests__/*.ts"],
    setupFiles: ["./vitest/vitest.setup.node.ts"],
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
