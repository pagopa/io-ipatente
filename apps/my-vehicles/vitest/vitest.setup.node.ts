/**
 * MSW configuration for Vitest (used as setupFile in vitest.config.node.ts)
 */
import { loadEnvConfig } from "@next/env";
import { afterAll, afterEach, beforeAll } from "vitest";

import { mswServerTest } from "../mocks/msw-server";

let mswServer: ReturnType<typeof mswServerTest>;
// Start server before all tests
beforeAll(() => {
  loadEnvConfig(process.cwd());
  // this done to correctly load the environment variables on msw handlers
  mswServer = mswServerTest();
  mswServer.listen({ onUnhandledRequest: "warn" });
});

//  Close server after all tests
afterAll(() => mswServer.close());

// Reset handlers after each test `important for test isolation`
afterEach(() => mswServer.resetHandlers());
