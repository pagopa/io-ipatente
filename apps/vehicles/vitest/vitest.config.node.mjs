import { loadEnv } from "vite";
import tsconfigPaths from "vite-tsconfig-paths";
import { configDefaults, defineConfig } from "vitest/config";

export default defineConfig({
  plugins: [tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...configDefaults.coverage.exclude,
        "**/__tests__",
        "src/generated/**",
        "src/app/api/auth/mocks/**",
        // FE
        "src/common/*",
        "src/hooks/*",
      ],
      ignoreEmptyLines: true,
      include: ["src/**/*.ts"],
      reporter: ["text", "html"],
      reportsDirectory: `${configDefaults.coverage.reportsDirectory}/backend`,
    },
    env: loadEnv("test", process.cwd(), ""),
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    include: ["./src/**/__tests__/*.ts"],
    setupFiles: ["./vitest/vitest.setup.node.ts"],
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
