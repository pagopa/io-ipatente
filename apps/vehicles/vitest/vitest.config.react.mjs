/// <reference types="vitest" />
/// <reference types="vite/client" />

import react from "@vitejs/plugin-react";
import tsconfigPaths from "vite-tsconfig-paths";
import { configDefaults, defineConfig } from "vitest/config";
import { loadEnv } from "vite";

export default defineConfig({
  plugins: [react(), tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...(configDefaults.coverage.exclude
          ? configDefaults.coverage.exclude
          : []),
        ".next/*",
        "mocks/*",
        "public/*",
        "*.config.js",
        "src/generated/**",
        "src/config/*",
        "src/types/*",
        "src/main.ts",
        "src/config.ts",
        "src/instrumentation.ts",
        "**/__mocks__/**",
        // BE
        "src/app/*",
        "src/lib/be/*",
      ],
      ignoreEmptyLines: true,
      reporter: ["text", "json-summary", "json"],
      reportsDirectory: `${configDefaults.coverage.reportsDirectory}/frontend`,
    },
    css: true,
    env: loadEnv("test", process.cwd(), ""),
    environment: "jsdom",
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    globals: true,
    include: ["./src/**/*.test.tsx"],
    setupFiles: "./vitest/vitest.setup.react.ts",
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
