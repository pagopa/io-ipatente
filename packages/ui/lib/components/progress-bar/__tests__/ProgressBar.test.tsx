import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import { ProgressBar } from "../ProgressBar";

const description = "sample-description";
const title = "sample-title";
const value = 20;
const total = 30;
const PROGRESS_BAR_TITLE_TEST_ID = "progress-bar-title";
const PROGRESS_BAR_DESCRIPTION_TEST_ID = "progress-bar-description";

describe("Test ProgressBar Component", () => {
  it("Should match the snapshot with all props", () => {
    const comp = render(
      <ProgressBar
        description={description}
        title={title}
        total={total}
        value={value}
      />,
    );
    const progressBarTitle = screen.queryByTestId(PROGRESS_BAR_TITLE_TEST_ID);
    const progressBarDescription = screen.queryByTestId(
      PROGRESS_BAR_DESCRIPTION_TEST_ID,
    );

    expect(progressBarTitle).toBeInTheDocument();
    expect(progressBarDescription).toBeInTheDocument();
    expect(comp).toMatchSnapshot();
  });
  it("Should match the snapshot without title and description", () => {
    const comp = render(<ProgressBar total={total} value={value} />);
    const progressBarTitle = screen.queryByTestId(PROGRESS_BAR_TITLE_TEST_ID);
    const progressBarDescription = screen.queryByTestId(
      PROGRESS_BAR_DESCRIPTION_TEST_ID,
    );

    expect(progressBarTitle).not.toBeInTheDocument();
    expect(progressBarDescription).not.toBeInTheDocument();
    expect(comp).toMatchSnapshot();
  });
});
