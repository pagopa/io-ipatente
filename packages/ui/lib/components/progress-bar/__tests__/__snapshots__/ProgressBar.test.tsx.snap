// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test ProgressBar Component > Should match the snapshot with all props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiStack-root css-14qxkfv-MuiStack-root"
      >
        <div
          class="MuiStack-root css-gmwslw-MuiStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-cjx14g-MuiTypography-root"
            data-testid="progress-bar-title"
          >
            sample-title
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-cjx14g-MuiTypography-root"
            data-testid="progress-bar-description"
          >
            sample-description
          </p>
        </div>
        <span
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="67"
          class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-1n4aybe-MuiLinearProgress-root"
          role="progressbar"
        >
          <span
            class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
            style="transform: translateX(-33.33333333333334%);"
          />
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiStack-root css-14qxkfv-MuiStack-root"
    >
      <div
        class="MuiStack-root css-gmwslw-MuiStack-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-cjx14g-MuiTypography-root"
          data-testid="progress-bar-title"
        >
          sample-title
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-cjx14g-MuiTypography-root"
          data-testid="progress-bar-description"
        >
          sample-description
        </p>
      </div>
      <span
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="67"
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-1n4aybe-MuiLinearProgress-root"
        role="progressbar"
      >
        <span
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
          style="transform: translateX(-33.33333333333334%);"
        />
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test ProgressBar Component > Should match the snapshot without title and description 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiStack-root css-14qxkfv-MuiStack-root"
      >
        <span
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="67"
          class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-1n4aybe-MuiLinearProgress-root"
          role="progressbar"
        >
          <span
            class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
            style="transform: translateX(-33.33333333333334%);"
          />
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiStack-root css-14qxkfv-MuiStack-root"
    >
      <span
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="67"
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-1n4aybe-MuiLinearProgress-root"
        role="progressbar"
      >
        <span
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
          style="transform: translateX(-33.33333333333334%);"
        />
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
